tests/virtctl/credentials.go

```diff
@@ -0,0 +1,280 @@
+	)
+
+	const (
+		sshKeySecretNamePrefix   = "test-secret-ssh-"
+		passwordSecretNamePrefix = "test-secret-password-"
+		userName                 = "root"
+	)
+
+	var (
+		cli kubecli.KubevirtClient
+
+		sshKeySecretName   string
+		passwordSecretName string
+		vm                 *kubevirtv1.VirtualMachine
+	)
+
+	BeforeEach(func() {
+		cli = kubevirt.Client()
+
+		sshKeySecretName = sshKeySecretNamePrefix + rand.String(6)
+		passwordSecretName = passwordSecretNamePrefix + rand.String(6)
+
+		vmi := libvmi.NewFedora()
+		vmi.Namespace = util.NamespaceTestDefault
+
+		vm = tests.NewRandomVirtualMachine(vmi, false)
+
+		vm.Spec.Template.Spec.AccessCredentials = []kubevirtv1.AccessCredential{{
+			SSHPublicKey: &kubevirtv1.SSHPublicKeyAccessCredential{
+				Source: kubevirtv1.SSHPublicKeyAccessCredentialSource{
+					Secret: &kubevirtv1.AccessCredentialSecretSource{
+						SecretName: sshKeySecretName,
+					},
+				},
+				PropagationMethod: kubevirtv1.SSHPublicKeyAccessCredentialPropagationMethod{
+					QemuGuestAgent: &kubevirtv1.QemuGuestAgentSSHPublicKeyAccessCredentialPropagation{
+						Users: []string{userName},
+					},
+				},
+			},
+		}, {
+			UserPassword: &kubevirtv1.UserPasswordAccessCredential{
+				Source: kubevirtv1.UserPasswordAccessCredentialSource{
+					Secret: &kubevirtv1.AccessCredentialSecretSource{
+						SecretName: passwordSecretName,
+					},
+				},
+				PropagationMethod: kubevirtv1.UserPasswordAccessCredentialPropagationMethod{
+					QemuGuestAgent: &kubevirtv1.QemuGuestAgentUserPasswordAccessCredentialPropagation{},
+				},
+			},
+		}}
+
+		vm, err := cli.VirtualMachine(util.NamespaceTestDefault).Create(context.Background(), vm)
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(func() {
+			err := cli.VirtualMachine(util.NamespaceTestDefault).Delete(context.Background(), vm.Name, &metav1.DeleteOptions{})
+			Expect(err).To(Or(
+				Not(HaveOccurred()),
+				Satisfy(errors.IsNotFound),
+			))
+		})
+
+		keySecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: sshKeySecretName,
+				OwnerReferences: []metav1.OwnerReference{{
+					APIVersion: kubevirtv1.VirtualMachineGroupVersionKind.GroupVersion().String(),
+					Kind:       kubevirtv1.VirtualMachineGroupVersionKind.Kind,
+					Name:       vm.Name,
+					UID:        vm.UID,
+					Controller: pointer.Bool(true),
+				}},
+			},
+			Data: map[string][]byte{
+				"key-file.pub": []byte(testKey1),
+			},
+		}
+
+		keySecret, err = cli.CoreV1().Secrets(util.NamespaceTestDefault).Create(context.Background(), keySecret, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(func() {
```

Comment:
Is this necessary? Aren't the secrets deleted by default on the namespace test cleanup?